### PR TITLE
send a hopeful disconnect packet from ACs and Interface

### PR DESCRIPTION
**Pull request by birarda**

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: 6b2987eef8c3abb3c44952c8a6d739e0c866cf5c available at: https://gist.github.com/bc88d83c6b48717a43f2<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6383.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6383.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: d932ba74fd75ef2244a165ad76c5e182fd9d4d67 available at: https://gist.github.com/bc88d83c6b48717a43f2<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6383.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6383.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: ea38c4cc49e90b98f9ccf92d9634a69658b5ebb6 available at: https://gist.github.com/bc88d83c6b48717a43f2<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6383.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6383.dmg

**Comment by nbq** [comment]
Tested this on both Mac and PC with new High Fidelity Server compile on Ubuntu 15.04 and it worked just fine. Both avatars drop the moment the client disconnects.

**Comment by PhilipRosedale** [comment]
@sethalves can you code review this?

File: domain-server/src/DomainServer.cpp [hunk 1]
```diff
@@ -1826,3 +1827,26 @@ void DomainServer::processPathQueryPacket(QSharedPointer<NLPacket> packet) {
...
+void DomainServer::processNodeDisconnectRequestPacket(QSharedPointer<NLPacket> packet) {
+    // This packet has been matched to a source node and they're asking not to be in the domain anymore
+    auto limitedNodeList = DependencyManager::get<LimitedNodeList>();
+    
+    const QUuid& nodeUUID = packet->getSourceID();
+    
+    qDebug() << "Received a disconnect request from node with UUID" << nodeUUID;
+    
+    if (limitedNodeList->nodeWithUUID(nodeUUID)) {
+        limitedNodeList->killNodeWithUUID(nodeUUID);
```
Comment (AndrewMeadows) [inline domain-server/src/DomainServer.cpp]: This does not block the PR, but...

If **killNodeWithUUID()** returned **true** on success then we wouldn't have to perform a double lookup.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 584e35e4ab1e4fb8d28d71533bbee3b9da84a2cd available at: https://gist.github.com/bc88d83c6b48717a43f2<br/><br/>Installers available for download at: <br/>Windows: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6383.exe<br/>OS X: http://s3.amazonaws.com/hifi-public/pr-builds/dev-interface-6383.dmg

**Pull request merged**